docs: vert.x example (#544)
awslabs/aws-advanced-jdbc-wrapper: examples/VertxExample/README.md

Create a Gradle Project with the following project hierarchy:
```
├───build.gradle
└───src
    └───main
        └───java
            └───com
                └───example
                    └───starter
                        ├───model
                        │   └───Example.java
                        │   └───Instance.java
                        └───VertxExampleApplication.java
```

You can also use the Vert.x initializer to generate a starter project:
1. Go to https://start.vertx.io/.
2. Select the Vert.x version, Java, and Gradle.
3. Add the following dependencies: 
   - JDBC Client
   - Vert.x Config
   - Vert.x Web
4. Generate the project.

## Step 2: Add the required Gradle dependencies
Your `build.gradle.kts` file should include the following dependencies:

```gradle
dependencies {
    implementation(platform("io.vertx:vertx-stack-depchain:4.4.2"))
    implementation("io.vertx:vertx-core")
    implementation("io.vertx:vertx-config")
    implementation("io.vertx:vertx-jdbc-client")
    implementation("io.vertx:vertx-web")
    implementation("com.fasterxml.jackson.core:jackson-databind:2.12.7.1")
    implementation("org.postgresql:postgresql:42.5.4")

>     implementation(project(":aws-advanced-jdbc-wrapper"))

Reviewer: The following question is a generic one and relates to all code examples. It make sense to refer to `":aws-advanced-jdbc-wrapper"` project when sample code is part of the the driver project. However, if a user wants to grab a sample code and make a separate sample app, such dependency should be replaced with a maven one.

implementation("software.amazon.jdbc:aws-advanced-jdbc-wrapper:latest")

Should we mention that in code examples?

```java
final JsonObject writeConfig = new JsonObject()
    .put("url", "jdbc:aws-wrapper:postgresql://db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com:5432/postgres")
    .put("driverClass", "software.amazon.jdbc.Driver")
    .put("user", "username")
    .put("password", "password")
    .put("testConnectionOnCheckout", "false")
    .put("testConnectionOnCheckin", "false")
    .put("idleConnectionTestPeriod", "30")
    .put("numHelperThreads", "10")
    .put("max_pool_size", 20);

final JsonObject readConfig = new JsonObject()
    .put("url", "jdbc:aws-wrapper:postgresql://db-identifier.cluster-ro-XYZ.us-east-2.rds.amazonaws.com:5432/postgres")
    .put("driverClass", "software.amazon.jdbc.Driver")
    .put("user", "username")
    .put("password", "password")
    .put("testConnectionOnCheckout", "false")
    .put("testConnectionOnCheckin", "false")
    .put("idleConnectionTestPeriod", "30")
    .put("numHelperThreads", "10")
    .put("max_pool_size", 20);

JDBCPool write;
JDBCPool read;
```

## Step 4: Set up Routing

> To set up the HTTP server, you can first create a router. This tutorial will only walk through creating one route, which will obtain the current instance ID, but the sample code will contain examples of different routes.

Reviewer 2: ```suggestion
To set up the HTTP server, you can first create a router. This tutorial will only walk through creating one route, which will obtain the instance ID of the current database. However, the sample code will contain examples of different routes.
```

Reviewer 2: Nit: Add clarity to what type of "instance" is being referred to.

> Note: this tutorial was written using the following technologies:
>    - AWS JDBC Wrapper 2.2.2
>    - PostgreSQL 42.5.4
>    - Java 8
>    - Vert.x 4.4.2

You will progress through the following sections:
1. Create a Gradle project
2. Add the required Gradle dependencies
3. Set up connection pools
4. Set up routing
5. Set up the start method
6. Run the application

## Step 1: Create a Gradle Project
Create a Gradle Project with the following project hierarchy:
```
├───build.gradle
└───src
    └───main
        └───java
            └───com
                └───example
                    └───starter
                        ├───model
                        │   └───Example.java
                        │   └───Instance.java
                        └───VertxExampleApplication.java
```

You can also use the Vert.x initializer to generate a starter project:
1. Go to https://start.vertx.io/.
2. Select the Vert.x version, Java, and Gradle.
3. Add the following dependencies: 
   - JDBC Client
   - Vert.x Config
   - Vert.x Web
4. Generate the project.

## Step 2: Add the required Gradle dependencies

> Your `build.gradle` file should include the following dependencies:

Reviewer 3: ```suggestion
Your `build.gradle.kts` file should include the following dependencies:
```

    implementation("io.vertx:vertx-core")
    implementation("io.vertx:vertx-config")
    implementation("io.vertx:vertx-jdbc-client")
    implementation("io.vertx:vertx-web")
    implementation("com.fasterxml.jackson.core:jackson-databind:2.12.7.1")
    implementation("org.postgresql:postgresql:42.5.4")
    implementation(project(":aws-advanced-jdbc-wrapper"))
}
```

## Step 3: Set up connection pools
Vert.x applications can use connection pools to handle requests. In this example, we use two connection pools, one to handle write requests and another to handle read requests. Vert.x uses C3P0 by default to manage database connections.

```java
final JsonObject writeConfig = new JsonObject()
    .put("url", "jdbc:aws-wrapper:postgresql://db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com:5432/postgres")
    .put("driverClass", "software.amazon.jdbc.Driver")
    .put("user", "username")
    .put("password", "password")
    .put("minSize", "10")
    .put("maxSize", "20")
    .put("testConnectionOnCheckout", "false")
    .put("testConnectionOnCheckin", "false")
    .put("idleConnectionTestPeriod", "30")
    .put("numHelperThreads", "10")

>     .put("max_pool_size", 16);

Reviewer 3: ```suggestion
    .put("max_pool_size", 20);
```

Reviewer 3: What is the difference between maxSize and max_pool_size?

Author: The maxSize and minSize properties were actually from [hibernate specific](https://www.mchange.com/projects/c3p0/#hibernate-specific) properties, they've been removed.

final JsonObject readConfig = new JsonObject()
    .put("url", "jdbc:aws-wrapper:postgresql://db-identifier.cluster-ro-XYZ.us-east-2.rds.amazonaws.com:5432/postgres")
    .put("driverClass", "software.amazon.jdbc.Driver")
    .put("user", "username")
    .put("password", "password")
    .put("minSize", "10")
    .put("maxSize", "20")
    .put("testConnectionOnCheckout", "false")
    .put("testConnectionOnCheckin", "false")
    .put("idleConnectionTestPeriod", "30")
    .put("numHelperThreads", "10")

>     .put("max_pool_size", 16);

Reviewer 3: ```suggestion
    .put("max_pool_size", 20);
```
